src/main/java/org/openstreetmap/atlas/checks/validation/areas/WaterbodyAndIslandSizeCheck.java

```diff
@@ -0,0 +1,336 @@
+    private final double islandMinimumArea;
+    private final double islandMaximumArea;
+    private final double isletMinimumArea;
+    private final double isletMaximumArea;
+
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            WATERBODY_MIN_INSTRUCTION, WATERBODY_MAX_INSTRUCTION, ISLAND_MIN_INSTRUCTION,
+            ISLAND_MAX_INSTRUCTION, ISLET_MIN_INSTRUCTION, ISLET_MAX_INSTRUCTION);
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public WaterbodyAndIslandSizeCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.waterbodyMinimumArea = (double) configurationValue(configuration,
+                "area.waterbody.minimum.meters", WATERBODY_MIN_AREA_DEFAULT);
+        this.waterbodyMaximumArea = (double) configurationValue(configuration,
+                "area.waterbody.maximum.kilometers", WATERBODY_MAX_AREA_DEFAULT);
+        this.islandMinimumArea = (double) configurationValue(configuration,
+                "area.island.minimum.meters", ISLAND_MIN_AREA_DEFAULT);
+        this.islandMaximumArea = (double) configurationValue(configuration,
+                "area.island.maximum.kilometers", ISLAND_MAX_AREA_DEFAULT);
+        this.isletMinimumArea = (double) configurationValue(configuration,
+                "area.islet.minimum.meters", ISLET_MIN_AREA_DEFAULT);
+        this.isletMaximumArea = (double) configurationValue(configuration,
+                "area.islet.maximum.kilometers", ISLET_MAX_AREA_DEFAULT);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Object must be a MultiPolygon Relation with at least one member
+        return (object instanceof Relation && IS_MULTIPOLYGON_WATER_RELATION.test(object)
+                && !((Relation) object).members().isEmpty())
+                // An Area with place=islet, place=island, place=archipelago, or natural=water tag
+                || (object instanceof Area && (TagPredicates.IS_WATER_BODY.test(object)
+                        || IS_ISLET.test(object) || IS_ISLAND.test(object)))
+                        // And has yet to be examined
+                        && !this.isFlagged(object.getIdentifier());
```

Comment:
Does the flagged object identifier only apply to Areas? I think that that makes sense because you shouldn't see a relation twice but I want to make sure that this is your intended effect.